src/Extension/Engine/SiteTreePublishingEngine.php

```diff
@@ -279,7 +279,10 @@ public function collectChanges($context)
             }
 
             // The page does not include the required extension, and it doesn't implement a Trigger
-            if (!$siteTree->hasExtension(PublishableSiteTree::class) && !($siteTree instanceof StaticPublishingTrigger)) {
+            if (!$siteTree->hasExtension(PublishableSiteTree::class)
```

Comment:
```suggestion
            if (
```
`PublishableSiteTree` implements `StaticPublishingTrigger` so there's no sense in checking for both.